### ISSUE TITLE
SemanticKITTI test miou lower than 73.3% only

### ISSUE THREAD

**Issue by cccuit** [Xiangxu-0103/FRNet#36]
![65](https://github.com/user-attachments/assets/e0cc11f2-4318-432c-a439-51bdee1ea171)

**Comment by Xiangxu-0103** [comment]
Hi @cccuit . For the test set, please refer to #13

**Comment by paladin1410** [comment]
@cccuit Hi, are you figuring out the the reason when training with SemanticKITTI dataset? I also try to train with SemanticKitti dataset and my result is even lower, mIOU =  **0.5902**. I did not modify any code, I just setup the SemanticKITTI dataset and start the training with python train.py configs/frnet/frnet-semantickitti_seg.py 
`+---------+--------+---------+------------+--------+--------+--------+-----------+--------------+--------+---------+----------+--------------+----------+--------+------------+--------+---------+--------+--------------+--------+--------+---------+
| classes | car    | bicycle | motorcycle | truck  | bus    | person | bicyclist | motorcyclist | road   | parking | sidewalk | other-ground | building | fence  | vegetation | trunck | terrian | pole   | traffic-sign | miou   | acc    | acc_cls |
+---------+--------+---------+------------+--------+--------+--------+-----------+--------------+--------+---------+----------+--------------+----------+--------+------------+--------+---------+--------+--------------+--------+--------+---------+
| results | 0.9396 | 0.4426  | 0.5553     | 0.4739 | 0.4077 | 0.5849 | 0.7075    | 0.0000       | 0.9550 | 0.5544  | 0.8256   | 0.0490       | 0.8721   | 0.5835 | 0.8732     | 0.6096 | 0.7568  | 0.5691 | 0.4538       | 0.5902 | 0.9175 | 0.6835  |
+---------+--------+---------+------------+--------+--------+--------+-----------+--------------+--------+---------+----------+--------------+----------+--------+------------+--------+---------+--------+--------------+--------+--------+---------+
06/13 13:14:44 - mmengine - INFO - Iter(val) [4071/4071]    car: 0.9396  bicycle: 0.4426  motorcycle: 0.5553  truck: 0.4739  bus: 0.4077  person: 0.5849  bicyclist: 0.7075  motorcyclist: 0.0000  road: 0.9550  parking: 0.5544  sidewalk: 0.8256  other-ground: 0.0490  building: 0.8721  fence: 0.5835  vegetation: 0.8732  trunck: 0.6096  terrian: 0.7568  pole: 0.5691  traffic-sign: 0.4538  miou: 0.5902  acc: 0.9175  acc_cls: 0.6835  data_time: 0.0283  time: 0.0613
`

**Comment by cccuit** [comment]
I previously contacted the original authors and received the following response:

Thank you for your interest in our work. For the results on the test set, we trained the model on the trainval split and applied a series of strategies such as fine-tuning based on pseudo labels for the test set, test-time augmentation (TTA), and other techniques.





------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "Xiangxu-0103/FRNet"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2025年6月16日(星期一) 上午10:03
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [Xiangxu-0103/FRNet] SemanticKITTI test miou lower than 73.3% only (Issue #36)



paladin1410 left a comment (Xiangxu-0103/FRNet#36)
 
@cccuit Hi, are you figuring out the the reason when training with SemanticKITTI dataset? I also try to train with SemanticKitti dataset and my result is even lower, mIOU =  0.5902. I did not modify any code, I just setup the SemanticKITTI dataset and start the training with python train.py configs/frnet/frnet-semantickitti_seg.py
 +---------+--------+---------+------------+--------+--------+--------+-----------+--------------+--------+---------+----------+--------------+----------+--------+------------+--------+---------+--------+--------------+--------+--------+---------+ | classes | car    | bicycle | motorcycle | truck  | bus    | person | bicyclist | motorcyclist | road   | parking | sidewalk | other-ground | building | fence  | vegetation | trunck | terrian | pole   | traffic-sign | miou   | acc    | acc_cls | +---------+--------+---------+------------+--------+--------+--------+-----------+--------------+--------+---------+----------+--------------+----------+--------+------------+--------+---------+--------+--------------+--------+--------+---------+ | results | 0.9396 | 0.4426  | 0.5553     | 0.4739 | 0.4077 | 0.5849 | 0.7075    | 0.0000       | 0.9550 | 0.5544  | 0.8256   | 0.0490       | 0.8721   | 0.5835 | 0.8732     | 0.6096 | 0.7568  | 0.5691 | 0.4538       | 0.5902 | 0.9175 | 0.6835  | +---------+--------+---------+------------+--------+--------+--------+-----------+--------------+--------+---------+----------+--------------+----------+--------+------------+--------+---------+--------+--------------+--------+--------+---------+ 06/13 13:14:44 - mmengine - INFO - Iter(val) [4071/4071]    car: 0.9396  bicycle: 0.4426  motorcycle: 0.5553  truck: 0.4739  bus: 0.4077  person: 0.5849  bicyclist: 0.7075  motorcyclist: 0.0000  road: 0.9550  parking: 0.5544  sidewalk: 0.8256  other-ground: 0.0490  building: 0.8721  fence: 0.5835  vegetation: 0.8732  trunck: 0.6096  terrian: 0.7568  pole: 0.5691  traffic-sign: 0.4538  miou: 0.5902  acc: 0.9175  acc_cls: 0.6835  data_time: 0.0283  time: 0.0613

**Comment by paladin1410** [comment]
@cccuit Thanks a lot for the response. My result mIOU=0.5902 seems to be for the evaluation set. Are you able to get the mIOU=0.687 for the evaluation set at the end the training as reported in the original paper?

**Comment by cccuit** [comment]
When I first started reproducing the paper, I wasn’t able to achieve the original results either. However, the network architecture proposed in the paper is effective, and I introduced some innovations that improved the performance metrics.

TTA:



------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "Xiangxu-0103/FRNet"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2025年6月16日(星期一) 上午10:19
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [Xiangxu-0103/FRNet] SemanticKITTI test miou lower than 73.3% only (Issue #36)



paladin1410 left a comment (Xiangxu-0103/FRNet#36)
 
@cccuit Thanks a lot for the response. My result mIOU=0.5902 seems to be for the evaluation set. Are you able to get the mIOU=0.687 for the evaluation set at the end the training as reported in the original paper?

**Comment by Xiangxu-0103** [comment]
@paladin1410 Thanks for your interest in our work. We suggest using 4 GPUs to better reproduce our results. Also, the results reported in the paper are evaluated under TTA, while the default config does not use TTA for evaluation. To use TTA, please refer to [GET_STARTED](https://github.com/Xiangxu-0103/FRNet/blob/master/docs/GET_STARTED.md).

**Comment by paladin1410** [comment]
@Xiangxu-0103 Thank you very much for the recommendation. I will try with 4 GPUs and TTA.

**Comment by paladin1410** [comment]
@cccuit  Would you mind sharing what changes you introduced? Thanks a lot.
